src/abstractions/src/services/EmptyServiceProvider.ts

```diff
@@ -1,10 +1,13 @@
-import { ServiceProvider } from './ServiceProvider';
+import { Constructor, ServiceProvider } from './ServiceProvider';
```

Comment:
This name, using a reserved keyword, might be confusing for users.  Possible rethink how the type is named to make it easier for those creating new to fit the type.

Reply:
Agreed.  I will refactor